OpenRA.Game/FileSystem/BagFile.cs

```diff
@@ -39,14 +39,14 @@ public BagFile(string filename, int priority)
 
 			// Build the index and dispose the stream, it is no longer needed after this
 			List<IdxEntry> entries;
-			using (var indexStream = GlobalFileSystem.Open(indexFilename))
+			using (var indexStream = Game.ModData.ModFiles.Open(indexFilename))
```

Comment:
`public BagFile(ModData modData, string filename, int priority)` and so on for the other package types.  They are all created from the same one place in the code, so this isn't a lot of work.